src/R/Components/Impl/ConnectionManager/Implementation/ConnectionInfo.cs

```diff
@@ -6,15 +6,16 @@
 namespace Microsoft.R.Components.ConnectionManager.Implementation {
     public class ConnectionInfo : IConnectionInfo {
         public string Name { get; set; }
+        public virtual string UserProvidedPath { get; set; }
```

Comment:
Connection overrides it. This one persists into settings, Connection also handles updating Path when UserProvidedPath is set. What's wrong with virtual properties?